compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -117,6 +117,11 @@ pub(crate) fn create_object_file(sess: &Session) -> Option<write::Object<'static
         "riscv32" => Architecture::Riscv32,
         "riscv64" => Architecture::Riscv64,
         "sparc64" => Architecture::Sparc64,
+        "avr" => Architecture::Avr,
+        "msp430" => Architecture::Msp430,
+        "hexagon" => Architecture::Hexagon,
+        "wasm32" => Architecture::Wasm32,
```

Comment:
This is likely going at the `object_file.write()` step, then `wasm32` will need to be removed from the list.